### ISSUE TITLE
The uri is reconstructed without all of it's parts

### ISSUE THREAD

**Issue by linchenko** [flutter/flutter#136927]
https://github.com/flutter/flutter/blob/25bed918b81beda4dec2f6a87515ce3630a81f72/packages/flutter/lib/src/widgets/app.dart#L1557
```
final Uri uri = routeInformation.uri;
navigator.pushNamed(
      Uri.decodeComponent(
        Uri(
          path: uri.path.isEmpty ? '/' : uri.path,
          queryParameters: uri.queryParametersAll.isEmpty ? null : uri.queryParametersAll,
          fragment: uri.fragment.isEmpty ? null : uri.fragment,
        ).toString(),
      ),
    );
```
Should be
```
navigator.pushNamed(
      Uri.decodeComponent(
       uri.toString(),
      ),
    );
```

Things like the host and scheme are dropped between the native and flutter layers

**Comment by linchenko** [comment]
@chunhtai

**Comment by chunhtai** [comment]
navigator.pushNamed only expects string after the path, either we throw it out before navigator.pushNamed or in  navigator.pushNamed. Is this causing any issue?

**Comment by linchenko** [comment]
Yes, before upgrading flutter versions we were receiving the entire value we passed from our native surfaces.
after the update we loose a lot of the uri

`ourScheme://some/route` => `/route`

to get the entire path we have to append an additional forward slash

`ourScheme:///some/route` => `/some/route`

We still loose access to the scheme though which we were using for specific handling for deep links

I would expect pushing a value from native to receive that same value in flutter. Another work around is adding an additional `/` to the entire uri

`/ourScheme://some/route` => `/ourScheme://some/route`

this just adds another thing to strip before we handle the route on the flutter surface.

**Comment by chunhtai** [comment]
Are you using a custom embedding?

I made the change a while back because all of our first party embedding were updated to send full uri, and I added this logic to remove the excessive part for backward compatibility.

**Comment by chunhtai** [comment]
I can't think of a way forward other than asking you to migrate your code to use Router API or just go_router so that you will still have access to RouteInformation.uri, or GoRouterState.uri.

**Comment by linchenko** [comment]
This broke backward compatibility for us haha. Calling [pushRoute](https://api.flutter.dev/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)pushRoute:) from FlutterViewController from the iOS codebase or equivalent on android runs through the logic of deconstructing the uri

**Comment by linchenko** [comment]
I still don't understand why the entire uri would be passed to be handled on the flutter layer

**Comment by chunhtai** [comment]
to handle deeplink request. For mobile app, there are use cases where deeplinks in different domain need to be handled differently.

**Comment by linchenko** [comment]
Right which is why I'm confused why I don't have access to the domain

**Comment by darshankawar** [comment]
Labeling it for further discussion and tracking as applicable.
